<launch>
    <param name="use_sim_time" value="false" />

    <!-- Main node -->
    <node name="orb_slam3" pkg="orb_slam3_ros" type="ros_mono_inertial" output="screen">
        <remap from="/camera/image_raw"     to="/camera_0"/>
        <remap from="/imu"                  to="/imu_0"/>

        <param name="voc_file"      type="string" value="$(find orb_slam3_ros)/orb_slam3/Vocabulary/ORBvoc.txt.bin"/>
        <param name="settings_file" type="string" value="$(find orb_slam3_ros)/config/Monocular-Inertial/test.yaml"/>

        <param name="world_frame_id"    type="string"   value="world" />
        <param name="cam_frame_id"      type="string"   value="camera_link" />
        <param name="imu_frame_id"      type="string"   value="imu_link" />
        <param name="enable_pangolin"   type="bool"     value="false" />


        <param name="save_map_path" type="string" value="/home/prostagma/orb_maps"/> <!-- Папка для сохранения -->

    </node>
    
    <!-- Visualization orb_slam3_with_imu -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find orb_slam3_ros)/config/my.rviz" output="screen"   launch-prefix="my-prime-run"/>
    
    <!-- Trajectory path -->
    <!-- <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_orb_slam3" output="screen" ns="orb_slam3_ros" >
        <param name="/target_frame_name" value="/world" />
        <param name="/source_frame_name" value="/imu_link" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node> -->
</launch>
